### ISSUE TITLE
32 bit build error

### ISSUE THREAD

**Issue by nyx** [thehydroimpulse/nanomsg.rs#97]
I experienced build errors on a 32 bit Linux machine:

```
cargo build
   Compiling link-config v0.1.1 (https://github.com/alexcrichton/link-config.git#9a30a0bf)
   Compiling nanomsg-sys v0.1.0 (https://github.com/thehydroimpulse/nanomsg.rs.git#771d7913)
   Compiling nanomsg v0.3.0 (https://github.com/thehydroimpulse/nanomsg.rs.git#771d7913)
/home/vagrant/.cargo/git/checkouts/nanomsg.rs-87a2823cf330bb35/master/src/lib.rs:405:17: 405:35 error: mismatched types:
 expected `u32`,
    found `u64`
(expected u32,
    found u64) [E0308]
/home/vagrant/.cargo/git/checkouts/nanomsg.rs-87a2823cf330bb35/master/src/lib.rs:405                 libnanomsg::NN_MSG,
                                                                                                     ^~~~~~~~~~~~~~~~~~
/home/vagrant/.cargo/git/checkouts/nanomsg.rs-87a2823cf330bb35/master/src/lib.rs:874:17: 874:35 error: mismatched types:
 expected `u32`,
    found `u64`
(expected u32,
    found u64) [E0308]
/home/vagrant/.cargo/git/checkouts/nanomsg.rs-87a2823cf330bb35/master/src/lib.rs:874                 libnanomsg::NN_MSG,
                                                                                                     ^~~~~~~~~~~~~~~~~~
error: aborting due to 2 previous errors
Could not compile `nanomsg`.
``` 

Looks like `libnanomsg::NN_MSG` is defined as a very large u64:

```
pub const NN_MSG: u64 = -1;
```

however `nn_recv` expects a `libc::size_t` which is apparently a `u32` on this platform:

```
pub fn nn_recv(socket: c_int, buf: *mut c_void, len: size_t, flags: c_int) -> c_int;
```

Not sure what the proper fix is yet.

```
$ cargo --version
cargo 0.0.1-pre-nightly (dfd4868 2015-02-06 22:55:29 +0000)
$ rustc --version
rustc 1.0.0-dev (725cc0646 2015-02-08 12:35:03 +0000)

Ubuntu 14.04.1 LTS
```

**Comment by blabaere** [comment]
@nyx NN_MSG is a constant from nanomsg, so it should be declared as 'isize'. Will fix this in a couple of hours.

**Comment by blabaere** [comment]
@nyx It's now fixed on branch master and branch stable_1.0
@thehydroimpulse I took the liberty of merging directly as it was a blocking one.

**Comment by thehydroimpulse** [comment]
@blabaere Yep, no worries! Feel free to do that. Not sure if you have access to the crates.io package?

**Comment by nyx** [comment]
this fix is confirmed on my end - thanks!

**Comment by blabaere** [comment]
By "access to the crates.io package", do you mean being able to publish a version ?
If yes, I did not even looked at how to do it yet.

**Comment by blabaere** [comment]
I tried to publish a new version of nanomsg-sys from the master, and failed because it depends on a source version of link-config. And of course the only available version on crates.io is the 0.1.1, that has been yanked ...

**Comment by thehydroimpulse** [comment]
@blabaere Yeah, that's what I meant. What do you think of dropping that dependency completely? I don't expect it to stop breaking even after 1.0 (because extensions won't be stable)

**Comment by blabaere** [comment]
@thehydroimpulse I'm fine with that but we should check that we are still able to produce the binaries on major platforms.

**Comment by blabaere** [comment]
@thehydroimpulse I managed to publish [nanomsg-sys](https://crates.io/crates/nanomsg-sys) 0.1.1 on crates.io but I'm not allowed to do so for nanomsg. Fixing this could be as simple as
```
thehydroimpulse@hal9001$ cargo login whatCratesIoStrangeApiKeyReally
thehydroimpulse@hal9001$ cargo owner --add blabaere
```

**Comment by thehydroimpulse** [comment]
@blabaere Done!

**Comment by blabaere** [comment]
@thehydroimpulse Thanks! I published version 0.3.1 from master and it works fine now.
